packages/@react-spectrum/combobox/src/ComboBox.tsx

```diff
@@ -224,6 +228,22 @@ const ComboBoxInput = React.forwardRef(function ComboBoxInput(props: ComboBoxInp
       )} />
   );
 
+  useEffect(() => {
+    let isLoading = loadingState === 'loading' || loadingState === 'filtering';
+
+    if (isLoading && !showLoading) {
+      // If loading is happening and the loading circle is not displayed, start timer to show loading circle
+      clearTimeout(timeout.current);
+      timeout.current = setTimeout(() => {
+        setShowLoading(true);
+      }, 1000);
```

Comment:
other iteration of this logic included:
- resetting timeout on every new filtering operation so that loading circle timing wouldn't start until user stopped typing (didn't really make sense when I thought about it more so I dropped it)

Reply:
Heh I was thinking the same thing, e.g. reset the timer every time the value changes. What made you think it didn't make sense?

Reply:
I was thinking of the case where the user is continuously typing in the field for more than 1 second and IMO that would be a valid time to show the loading circle to let the user know that filtering/loading is happening.

Reply:
hmm yeah. on the flip side, the original request for the first letter has long been canceled by that point and we're really waiting on the request for the last letter you typed. 🤷 

anyway, I asked spectrum about it in slack. we'll see what they think.